ml-agents/mlagents/trainers/components/reward_signals/curiosity/model.py

```diff
@@ -0,0 +1,179 @@
+import tensorflow as tf
+from mlagents.trainers.models import LearningModel
+
+
+class CuriosityModel(object):
+    def __init__(
+        self,
+        policy_model: LearningModel,
+        encoding_size: int = 128,
+        learning_rate: float = 3e-4,
+    ):
+        """
+        Creates the curiosity model for the Curiosity reward Generator
+        :param policy_model: The model being used by the learning policy
+        :param encoding_size: The size of the encoding for the Curiosity module
+        :param learning_rate: The learning rate for the curiosity module
+        """
+        self.encoding_size = encoding_size
+        self.policy_model = policy_model
+        encoded_state, encoded_next_state = self.create_curiosity_encoders()
+        self.create_inverse_model(encoded_state, encoded_next_state)
+        self.create_forward_model(encoded_state, encoded_next_state)
+        self.create_loss(learning_rate)
+
+    def create_curiosity_encoders(self):
```

Comment:
Can you add type annotations to some of these methods? This one is just `def create_curiosity_encoders(self) -> None:`